pkg/apis/api.go

```diff
@@ -0,0 +1,27 @@
+// +nirvana:api=descriptors:"Descriptor"
+
+package apis
+
+import (
+	v1alpha1 "github.com/caicloud/nirvana-practice/pkg/apis/v1alpha1/descriptors"
+
+	middleware "github.com/caicloud/go-common/nirvana/middleware"
+	def "github.com/caicloud/nirvana/definition"
+	"github.com/caicloud/nirvana/log"
+)
+
+// Descriptor returns a combined descriptor for APIs of all versions.
+func Descriptor() def.Descriptor {
+	return def.Descriptor{
+		Description: "APIs",
+		Path:        "/api",
+		Middlewares: []def.Middleware{
+			middleware.Reqlog(log.DefaultLogger()),
```

Comment:
Use the log middleware in [go-common](https://github.com/caicloud/go-common/blob/master/nirvana/middleware/middleware.go)

Reply:
go-common is a private repo and this is a public repo

Reply:
Have copy the log middleware from go-common to this repo.

Reply:
Maybe we can just change go-common to a public repo?

My concern is that once we start advocating this practice, then people will start to use this copied version of log package, instead of the one in go-common, making it difficult to maintain.

Reply:
> Maybe we can just change go-common to a public repo?

I don't like that idea, at least not now. It'll make things more complicated.

Instead, why don't we add some of the more mature features from go-common to Nirvana? One thing we should definitely think about is to implement some built-in middleware in Nirvana. `reqlog` is obvious. The `metrics` plugin would also use a middleware implementation (you don't usually want request metrics on the `/metrics` API).

Reply:
SGTM, we can move the plugins into nirvana core, or create another dedicated `nirvana-plugins` repo.